### PR TITLE
Feature/tamil

### LINKED ISSUE #8: Create Tamil tenant
## Database changes:
1. Table `language`:
```sql 
INSERT INTO language (code, name, locale) VALUES ("ta", "Tamil", "ta_IN");
```
2. Table `instance`:
```sql
INSERT INTO instance (language_id, name,subdomain) VALUES ((SELECT id FROM language WHERE code="ta"),"Tamil", "ta");
``` 
3. roles and permissions (see below)
4. taxonomy 

## File changes:
1. Add language to php config in Dockerfile https://github.com/serlo/serlo.org/blob/7f26740906d6e42de6058717a585df4d6da5e47f/packages/public/server/docker/php/Dockerfile#L7-L11
2. add folders and files in [src/lang](https://github.com/serlo/serlo.org/tree/master/packages/public/server/src/lang) and [src/lang/routes](https://github.com/serlo/serlo.org/tree/master/packages/public/server/src/lang/routes)
3. add tenant branding configs to src/config/autoload/global https://github.com/serlo/serlo.org/blob/7f26740906d6e42de6058717a585df4d6da5e47f/packages/public/server/src/config/autoload/global.php#L41-L72
4. Tenant specific secrets (e.g. tracking)
5. Add the instance with subject config (at least `/community`) to [src/module/Subject/config/instances.config.php](https://github.com/serlo/serlo.org/blob/7f26740906d6e42de6058717a585df4d6da5e47f/packages/public/server/src/module/Subject/config/instances.config.php)
## On serlo.org (After everything is set up!)
1. Add license through backend (/license/manage)
2. Add subjects through backend (/taxonomy/term/organize-all)

### LINKED ISSUE #9: allow hindi (and other) letters in url?!
e.g. https://hi.serlo.org/-112518

### LINKED ISSUE #10: problem with Hindi URLs
we tried to change the URL of https://hi.serlo.org/--- to https://hi.serlo.org/जुड़-जाना but the hindi letters have been replaced by "---"

### PR DESCRIPTION
**Pull request by Knorrke**
### Fixed:
Allow UTF-8 in Aliases - fixes #10, closes #9 

### Added:
Enable Hindi tenant - closes #8 

For this to work, some additional steps are needed (see #8):
1. Run the sql script in the database to set up the necessary data for a language tenant
2. Afterwords: (ping @jakobwes) Add a license (/license/manage), subjects (/taxonomy/term/organize-all), and pages (/pages). The configured subjects are `community`, `இலக்கணம்`, `வரலாறு`, and one page should have the url `/serlo`

### CODE REVIEW

**Comment by Knorrke** [comment]
```sql
SET @langCode = "ta";
SET @langLocale = "ta_IN";
SET @langName = "Tamil";
/*
 * Setup basic tenant information
 */
-- Insert language
INSERT INTO language (code, name, locale) VALUES (@langCode, @langName, @langLocale);

-- Insert instance
INSERT INTO instance (language_id, name,subdomain) VALUES ((SELECT id FROM language WHERE code=@langCode),@langName, @langCode);

SET @instance = (SELECT id FROM instance i WHERE i.subdomain="ta");

/*
 * Setup role system
 */
SET @langLower = LOWER(@langName)

-- Insert roles
INSERT INTO role(name) VALUES 
(@langLower + "_reviewer"),
(@langLower + "_helper"),
(@langLower + "_admin"),
(@langLower + "_horizonhelper"),
(@langLower + "_moderator"),
(@langLower + "_ambassador"),
(@langLower + "_langhelper"),
(@langLower + "_langadmin");

-- Add inheritance of roles
INSERT INTO role_inheritance (role_id, child_id) VALUES 
((SELECT id FROM role WHERE name=(@langLower + "_reviewer")),(SELECT id FROM role WHERE name="login"))),
((SELECT id FROM role WHERE name=(@langLower + "_helper")),(SELECT id FROM role WHERE name=(@langLower + "_reviewer"))),
((SELECT id FROM role WHERE name=(@langLower + "_admin")),(SELECT id FROM role WHERE name=(@langLower + "_helper"))),
((SELECT id FROM role WHERE name=(@langLower + "_moderator")),(SELECT id FROM role WHERE name=(@langLower + "_admin"))),
((SELECT id FROM role WHERE name=(@langLower + "_ambassador")),(SELECT id FROM role WHERE name=(@langLower + "_moderator"))),
((SELECT id FROM role WHERE name=(@langLower + "_langhelper")),(SELECT id FROM role WHERE name=(@langLower + "_ambassador"))),
((SELECT id FROM role WHERE name=(@langLower + "_langadmin")),(SELECT id FROM role WHERE name=(@langLower + "_langhelper"))),
((SELECT id FROM role WHERE name="sysadmin"),(SELECT id FROM role WHERE name=(@langLower + "_langadmin")));


-- Copy non-global Permissions from english roles 
INSERT INTO instance_permission (permission_id, instance_id)
SELECT DISTINCT i.permission_id, @instance
FROM role_permission p JOIN role r ON p.role_id = r.id JOIN instance_permission i ON p.permission_id = i.id
WHERE r.name LIKE "english_%" AND i.instance_id IS NOT NULL;

-- Copy global Permissions from english roles
INSERT INTO instance_permission (permission_id, instance_id)
SELECT DISTINCT i.permission_id, NULL
FROM role_permission p JOIN role r ON p.role_id = r.id JOIN instance_permission i ON p.permission_id = i.id
WHERE r.name LIKE "english_%" AND i.instance_id IS NULL;

-- Connect the created permissions to the same roles as in english
INSERT INTO role_permission(role_id, permission_id)
SELECT rt.id, it.id
FROM role re /* english roles */
  JOIN role rt ON rt.name = REPLACE(re.name, "english", @langLower) /* corresp. tamil roles */
  JOIN role_permission pe ON re.id = pe.role_id 
  JOIN instance_permission ie ON ie.id = pe.permission_id, /* english permissions */
  instance_permission it /* this will be used for corresp. tamil permissions */
WHERE 
  re.name LIKE "english_%" AND 
  it.permission_id = ie.permission_id /* permission of english and tamil should be the same */
  AND NOT ie.id = it.id /* should be different elements */
  AND (it.instance_id IS NULL AND ie.instance_id IS NULL /* global permission */ 
       OR it.instance_id = @instance /* tamil permission */
);

/*
 * Setup taxonomy root
 */

-- Create the taxonomy root and subject if not existing yet
INSERT INTO taxonomy(type_id, instance_id)
SELECT t.id, @instance
FROM type t
WHERE (t.name = "root" OR t.name = "subject") AND NOT t.id IN (SELECT type_id FROM taxonomy WHERE instance_id = @instance)

SET @rootTax = (SELCECT id FROM taxonomy tax JOIN type t ON tax.type_id = t.id WHERE t.name="root" AND tax.instance_id = @instance);
SET @subjectTax = (SELCECT id FROM taxonomy tax JOIN type t ON tax.type_id = t.id WHERE t.name="subject" AND tax.instance_id = @instance);

-- Create root term and connect to taxonomy
INSERT INTO term(instance_id, name) 
VALUES (@instance, "root");

SET @rootTerm = LAST_INSERT_ID();

INSERT INTO uuid(trashed, discriminator)
VALUES (0, "taxonomyTerm");

SET @rootTermTax = LAST_INSERT_ID();

INSERT INTO term_taxonomy(id, taxonomy_id, term_id, parent_id) VALUES
  (@rootTermTax, @rootTax, @rootTerm, NULL);
```

**Comment by inyono** [comment]
- Add GA tracking code for ta.serlo.org and add it to the secrets

File: packages/public/server/src/module/Subject/config/instances.config.php [hunk 1]
```diff
@@ -360,7 +362,21 @@
                         'applet',
                     ],
                 ],
-                'test area' => [
```
Comment (inyono) [inline packages/public/server/src/module/Subject/config/instances.config.php]: Did you intentionally remove the test area of spanish and hindi tenant?

**Comment by inyono** [comment]
@inyono: - [ ] deploy static assets

File: packages/public/server/src/module/Subject/config/instances.config.php (hunk 1 above)
Comment (Knorrke) [reply packages/public/server/src/module/Subject/config/instances.config.php]: yes, spanish and hindi tenant dont have a taxonomy yet so this is not set up yet and it was removed in https://en.serlo.org/test-area and https://de.serlo.org/testbereich, so I figured we don't need this in new tenants.

**Pull request merged**